src/Node.ipp

```diff
@@ -17,15 +17,24 @@ Publisher<T> Node::create_publisher(CanardPortID const port_id,
 }
 
 template <typename T>
-bool Node::subscribe(OnTransferReceivedFunc func)
+Subscription<T> Node::create_subscription(CanardPortID const port_id,
+                                          CanardMicrosecond const rx_timeout_usec,
+                                          std::function<void(T const &)> on_receive_cb)
```

Comment:
Same comment that I left below applies in this case as well: replace std::function with a native lambda to reduce the number of moving parts at runtime.

Reply:
Not sure how to do this without creating an extremely awkward `create_subscription` call, i.e.

```C++
auto sub = node_hdl.create_subscription<Heartbeat_1_0<>, std::function<void(Heartbeat_1_0<> const &)>, std::function<void(void)>>(...);
```

Reply:
You don't need the other two template arguments because they will be deduced automatically. You would go like:

```c++
auto sub = node_hdl.create_subscription<Heartbeat_1_0<>>(
    port_id,
    rx_timout_usec,
    [](const Heartbeat_1_0<>& msg){ ... });
```

The implementation could be like:

```c++
template <typename T, typename OnReceiveCb>
Subscription<T> Node::create_subscription(CanardPortID const port_id,
                                          CanardMicrosecond const rx_timeout_usec,
                                          OnReceiveCb&& on_receive_cb)
{
    auto sub = std::make_shared<impl::Subscription<T>>(std::forward<OnReceiveCb>(on_receive_cb),
                                                       [this, port_id]() { unsubscribe_message(port_id); });
   ...
}
```

Reply:
Thank you :rocket: :bow: 

Done in 7c7c05383b1ad8000752d434279fba80f96f4e33 !